Assets/Scripts/StandardSamples/UI/Common/SampleMenu.cs

```diff
@@ -0,0 +1,316 @@
+        /// </summary>
+        protected virtual void Update()
+        {
+            // If nothing is selected, set the selected (focused) control to be
+            // UIFirstSelected (if that field has been set).
+            SetSelected();
+        }
+
+        /// <summary>
+        /// Shows the SampleMenu. If overriding, make sure to call this base
+        /// implementation first.
+        /// </summary>
+        public virtual void Show()
+        {
+            Log($"Show() started");
+
+            // Don't do anything if already showing.
+            if (!Hidden) return;
+
+            if (null == UIParent)
+            {
+                Log($"UIParent for is null for some reason.");
+            }
+            else
+            {
+                UIParent.SetActive(true);
+            }
+
+            // Make the gameObject active. This should make all the relevant
+            // items in the sample menu visible and interactable. 
+            if (null == gameObject)
+            {
+                Log($"GameObject for is null for some reason.");
+                return;
+            }
+            else
+            {
+                gameObject.SetActive(true);
+            }
+
+            // Set the selected control to be the first selected element, so
+            // long as that element can actually be selected.
+            SetSelected();
+
+            // Flag as showing.
+            Hidden = false;
+
+            Log($"Show() completed");
+        }
+
+        /// <summary>
+        /// Hides the SampleMenu. If overriding, make sure to call this base
+        /// implementation first.
+        /// </summary>
+        public virtual void Hide()
+        {
+            Log($"Hide() started");
+
+            // Don't do anything if already hidden.
+            if (_hidden.HasValue && _hidden.Value) return;
+
+            if (null == UIParent)
+            {
+                Log($"UIParent is null for some reason.");
+            }
+            else
+            {
+                UIParent.SetActive(false);
+            }
+
+            // Make the gameObject inactive. This should make all the relevant 
+            // items in the sample menu invisible and non-interactable.
+            gameObject.SetActive(false);
+
+            // Flag as hidden.
+            Hidden = true;
+
+            Log($"Hide() completed");
+        }
+
+        /// <summary>
+        /// Sets the focused control to UIFirstSelected, assuming it is both
+        /// defined and active in the hierarchy.
+        /// </summary>
+        private void SetSelected()
+        {
+            // Controller: Detect if nothing is selected and controller input detected, and set default
+            if (UIFirstSelected == null 
+                || UIFirstSelected.activeSelf == false
+                || UIFirstSelected.activeInHierarchy == false
+                || EventSystem.current == null || EventSystem.current.currentSelectedGameObject != null
```

Comment:
Perhaps we could check if the currentSelectedGameObject is not null, but also its GameObject is set to be inactive. Like if there's a menu with a button, and we navigate to this scene while it's still set.

Maybe Unity is smart about this and already manages unsetting the currentSelectedGameObject if it becomes hidden.

Reply:
I'm fairly certain that Unity is smart enough about this. Meaning that if a gameObject is set as inactive, then Unity knows to remove it as the `currentSelectedGameObject`. Of course, this is an assumption on my part. 

While this function appears to be new, it is actually transplanted as it existed in one of the other `Menu` classes. Most of the existing `Menu` classes implement a function similar to this one, but this implementation is a reflection of the implementation that I found to be most complete.

Reply:
Study shows that an object being inactive does not unset itself. We discussed checking for its activeness if it is set.